cpp/cppsdk/gsdk.cpp

```diff
@@ -309,6 +309,17 @@ namespace Microsoft
                             }
                         }
 
+                        if (sessionConfig.isMember("metadata"))
+                        {
+                            Json::Value sessionMetadata = sessionConfig["metadata"];
+                            for (Json::ValueIterator i = sessionMetadata.begin(); i != sessionMetadata.end(); ++i)
+                            {
+                                if ((*i).isString())
+                                {
+                                    m_configSettings[i.key().asCString()] = (*i).asCString();
```

Comment:
>m_configSettings [](start = 36, length = 16)

It seems strange that we flatten the heartbeat so it is the same if it is in metadata or in the parent node. I am not sure of the goal of the metadata node so it is hard to say if this is the correct behavior.

Reply:
Looks like I read the PRs in the wrong order. I think I understand the goal now and it still seems strange to flatten and could lead to bugs if there is a metadata value that has the same name as another config value.

Maybe we could have a new map<string,string> Metadata or maybe there is another solution. Let's discuss because we can't change it later.

Same comment applies to c#

---
In reply to: [323535059](https://github.com/PlayFab/gsdk/pull/37#discussion_r323535059) [](ancestors = 323535059)